### ISSUE TITLE
Examples are excluded from workspace

### ISSUE THREAD

**Issue by oceantume** [gbj/leptos#28]
Examples are excluded from the workspace and this makes rust-analyzer skip them on my machine. Do you use a special setup to circumvent that, or is this an oversight? While I don't have much experience with setting up rust projects, I feel like examples should simply be members of the workspace.

**Comment by gbj** [comment]
I'm open to changing it; the reason they're excluded at present is so that the examples have the possibility of defining their own `release` profiles. If you move the examples from being excluded to being members of the workspace, it fires one of these warnings for each of them
```
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /Users/gjohnston/Documents/Projects/leptos-main/leptos/examples/todomvc-ssr/todomvc-ssr-client/Cargo.toml
workspace: /Users/gjohnston/Documents/Projects/leptos-main/leptos/Cargo.toml
```

This suggests to me that we'd need to set a single profile for all the examples, but there are different settings that make sense for them. Particularly in the SSR + hydration examples, for `release` builds it's nice to optimize for size on the Wasm frontend and for speed on the server. It's not a big deal as these are just examples.

If anyone knows how to get around this problem, I'm open to hearing it.

I tend to just open the examples in their own editor window so I don't notice the lack of rust-analyzer support but that sounds like a pretty good argument for including them in the workspace. When you say rust-analyzer skips them, does that mean it just doesn't run at all for the examples? That's not ideal obviously.

**Comment by oceantume** [comment]
> When you say rust-analyzer skips them, does that mean it just doesn't run at all for the examples? That's not ideal obviously.

Indeed, that's what I mean. I'm not really knowledgeable in that area either but I'm interested in this project and I may try to play with the config a bit if I end up trying to contribute.

**Comment by akesson** [comment]
Sorry for the warning, it's due to my PR #22. The solution for the warning is to add an empty `[workspace]` at the top of the Cargo.toml that is not included in the workspace (I'll raise a PR for it as soon as I have some time for it). 

@gbj, is this right? I'm basing myself on what I see in the hackernews example.

In the currently workspaced Leptos libraries, there are a set of mutually exclusive features: `ssr`, `csr` and `hydrate`. 

When coding the app, it is done in a way that is compatible with all three. 

When compiling the client, either `csr` or `hydrate` can be used. When compiling the server `ssr` needs to be enabled (if you want it).

This means that the server and client needs conflicting features and if included in the workspace there would be errors due to these conflicts.

It can potentially be solved by Cargo's [Feature resolver 2](https://doc.rust-lang.org/cargo/reference/features.html#feature-resolver-version-2) but would need more investigation.

**Comment by akesson** [comment]
Sorry, not keeping up with the changes here... On latest main branch I don't get the warning and I see that @gbj already added examples as excluded (which has the same effect as adding `[workspace]` to the example projects' Cargo.toml.

**Comment by gbj** [comment]
I've decided to include the examples in the workspace and just define one profile at the workspace root... it's really not a big deal that we can't have a separate profile for the example servers.

**Comment by adsick** [comment]
@gbj can you explain please? Right now I see examples as excluded from the workspace. More on that in my feedback here: #2141.
